### PR TITLE
Fix organization article ownership code

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Documentation Update

## Description

After recent changes we lost a bit of organization logic that wasn't tested, so this is my attempt at fixing the bug. Thanks @Zhao-Andy for pointing me in the right direction :D

I plan to refactor the whole "article update" logic so that both the API and the frontend have the same capabilities and there's a central place where this logic resides, but this is for a future PR

## Related Tickets & Documents

Closes https://github.com/thepracticaldev/dev.to/issues/2871

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -142,12 +142,24 @@ def update
...
-                   @article.to_json(only: [:id], methods: [:current_state_path])
-                 else
-                   @article.errors.to_json
-                 end
+    updated = @article.update(article_params_json.merge(edited_at: edited_at_date))
+    Notification.send_to_followers(@article, "Published") if updated && @article.published && @article.saved_changes["published_at"]&.include?(nil)
+
+    respond_to do |format|
+      format.html do
+        # NOTE: destination is used by /dashboard/organization when it re-assigns an article
```
Comment (benhalpern) [inline app/controllers/articles_controller.rb]: Ah. Good call.

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -213,25 +225,39 @@ def article_params
...
+    allowed_params = if params["article"]["version"] == "v1"
+                       %i[body_markdown organization_id]
+                     else
+                       %i[
+                         title body_markdown main_image published description
+                         tag_list organization_id canonical_url series collection_id
+                       ]
+                     end
+
+    # NOTE: the organization logic is still a little counter intuitive but this should
```
Comment (benhalpern) [inline app/controllers/articles_controller.rb]: Agreed. But this looks good

File: app/controllers/articles_controller.rb [hunk 3]
```diff
@@ -213,25 +225,39 @@ def article_params
     params.require(:article).permit(modified_params)
   end
 
+  # TODO: refactor all of this update logic into the Articles::Updater possibly,
```
Comment (benhalpern) [inline app/controllers/articles_controller.rb]: Yup, exactly why we have objects like this. No clean reason for the logic to live in the controllers.

**Comment by philnash** [comment]
Is this deployed? Should I be able to update an author on an org post safely again?

**Comment by rhymes** [comment]
@philnash it has been deployed, please test it out with a draft article and backup its Markdown before just in case. Thanks for the patience

**Pull request merged**